var/spack/repos/builtin/packages/ascent/package.py

```diff
@@ -102,29 +105,54 @@ class Ascent(Package, CudaPackage):
     depends_on("mpi", when="+mpi")
     depends_on("py-mpi4py", when="+mpi+python+shared")
 
+    #######################
+    # BabelFlow
+    #######################
+    depends_on('babelflow@develop', when='+babelflow+mpi')
+    depends_on('pmt@develop', when='+babelflow+mpi')
```

Comment:
So this isn't needed for `+babelflow~mpi`? I think your `spec['babelflow'].prefix` below will crash in this case.

Reply:
thanks for noticing this, I moved babelflow logic to be mpi dependent.